### ISSUE TITLE
Using beam search on validation set

### ISSUE THREAD

**Issue by Zhaofeng-Wu** [OpenNMT/OpenNMT-py#1503]
Skimming through the code, it looks like the validation during training is not done with beam search + BLEU. However since that is the ultimate way of evaluating the model (i.e. we do beam search + BLEU on the test set), wouldn't it be more accurate if we do the exact same thing to validate the model? It is going to be slower though.

**Comment by vince62s** [comment]
it's in the project list and someone attempted a PR but not exactly suited.
If you feel like contributing, super welcome.
